src/main/studio/ui/StudioPanel.java

```diff
@@ -878,7 +878,9 @@ public static void settings() {
             StudioOptionPane.showMessage(activePanel.frame, "Look and Feel was changed. " +
                     "New L&F will take effect on the next start up.", "Look and Feel Setting Changed");
         }
-
+        Font font = new Font(dialog.getFontName(), Font.PLAIN, dialog.getFontSize());
+        Config.getInstance().setFont(font);
+        activePanel.editor.getPane().setFont2(font);
```

Comment:
1) There is `CONFIG` variable - the code above use it instead of `Config.getInstance()`
2) Why didn't you modify the font for other opened panes and frames ?
3) Font size of line numbers are not changed. So it will be inconsistent result on previously opened panes and new panes.

Reply:
I tried this:
```
        CONFIG.setFont(Config.FONT_EDITOR, font);
        for(StudioPanel panel: allPanels) {
            panel.editor.getPane().setTextAreaFont(font);
        }
```
and
```
    public void setTextAreaFont(Font font) { //don't call this setFont, it leads to an error
        textArea.setFont(font);
        scrollPane.getGutter().setLineNumberFont(font);
    }
```
But this results in the currently open tab getting the updated line numbers and spacing, while the other tabs still having the old size for the numbers and the old line spacing but with the actual font of the text updated. Any idea what I'm doing wrong?

Reply:
You iterate over panels (which are frames) but not tabs. I think you need to add the logic into `refreshEditorSetting()`